### PR TITLE
[FIX] Make CMakeFiles work with Windows Visual Studio

### LINKED ISSUE #812: [BUG] CMake does not work with Visual Studio
Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

CCExtractor version (using the --version parameter preferably) : 0.85

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/CCExtractor/ccextractor/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/CCExtractor/ccextractor/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.
- [X] I have used the latest available version of CCExtractor to verify this issue exists.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never used CCExtractor.
- [X] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [ ] I am an active contributor to CCExtractor.

**Necessary information**
- Is this a regression (did it work before)? [ ] NO | [ ] YES - *please specify the last known working version*
- What platform did you use? [X] Windows - [ ] Linux - [ ] Mac
- What where the used arguments? N/A

**Video links**

N/A

**Additional information**

CMake does not work with Visual Studio. Following errors raises when I was trying to compile it using the .sln file generated by CMake:
```
1>base_encoding.c
1>d:\gitprojects\ccextractor\src\zlib\zconf.h(452): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
1>bitstream.c
1>mp4.c
1>d:\gitprojects\ccextractor\src\lib_ccx\stdintmsc.h(90): error C2371: 'int_fast16_t': redefinition; different basic types
1>d:\gitprojects\ccextractor\src\lib_ccx\stdintmsc.h(94): error C2371: 'uint_fast16_t': redefinition; different basic types
1>os_divers.c
1>D:\GitProjects\ccextractor\src\gpacmp4\os_divers.c(57): fatal error C1083: Cannot open include file: 'sys/time.h': No such file or directory
1>os_file.c
1>D:\GitProjects\ccextractor\src\gpacmp4\os_file.c(44): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
2>cl : Command line error D8021: invalid numeric argument '/Wno-write-strings'
2>Done building project "ccextractor.vcxproj" -- FAILED.
```
CMake version used: 3.7.2

However, windows\ccextractor.sln works.

### PR DESCRIPTION
**Pull request by harrynull**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [ ] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [X] I am an active contributor to CCExtractor.

---

Make CMakeFiles work with Windows Visual Studio

Fix #812

### CODE REVIEW

File: src/lib_ccx/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,14 @@
 cmake_policy (SET CMP0037 NEW)
 
-set (CMAKE_C_FLAGS "-O0 -Wall -g -std=gnu99")
+if(MSVC)
+    set (CMAKE_C_FLAGS "-W3 /wd4005 /wd4996")
+else (MSVC)
+    set (CMAKE_C_FLAGS "-Wall -g -std=gnu99")
```
Comment (anshul1912) [inline src/lib_ccx/CMakeLists.txt]: removed O0 from original linux compilation, if it is unintentional please add it again, or provide reason

Comment (harrynull) [reply src/lib_ccx/CMakeLists.txt]: @anshul1912 Doesn't -O0 mean turn off optimization? Why do we want to turn off optimization?

Comment (anshul1912) [reply src/lib_ccx/CMakeLists.txt]: So you want to turn on optimization thats a good reason for removing.

**Comment by anshul1912** [comment]
Can you please check why test case are failing for windows

**Comment by harrynull** [comment]
They always fail in Windows. Example: https://sampleplatform.ccextractor.org/test/1072

**Pull request merged**